packages/transaction-controller/src/TransactionController.ts

```diff
@@ -410,19 +418,22 @@ export class TransactionController extends BaseController<
    * @param options - The controller options.
    * @param options.getNetworkState - Gets the state of the network controller.
    * @param options.onNetworkStateChange - Allows subscribing to network controller state changes.
-   * @param options.getProvider - Returns a provider for the current network.
+   * @param options.getProviderAndBlockTracker - Returns a provider and blockTracker for the current network.
    * @param config - Initial options used to configure this controller.
    * @param state - Initial state to set on this controller.
    */
   constructor(
     {
       getNetworkState,
       onNetworkStateChange,
-      getProvider,
+      getProviderAndBlockTracker,
```

Comment:
The advantage of using a proxy is that we can pass in a single provider/blocktracker and it's always pointed at the correct network. So we don't need a "getter" method like this. Passing in the provider/blocktracker instances directly also means we don't need to bring in this swappable object proxy library/type

Reply:
right! Makes sense!

Reply:
We'd still be passing in the proxies though correct? So in the end we _would_ be using the types from `swappable-obj-proxy`, we just wouldn't have to import them.

Reply:
Hmm.... yeah that's a good point. I guess I was thinking that we could type these as non-proxy providers, and that would be valid from a type perspective because we aren't using the proxy functionality here. I don't know if that makes sense though.

Reply:
Yeah, I'm not sure about that... it might be good to reflect reality considering that proxy objects technically have an additional method to change the underlying object.

Reply:
Ideally the network controller would expose proxies that can't be modified. We really only want the network controller to have that capability. Not sure the best way to do that though.

Reply:
Well, that's a good point. Maybe we hide the `setProxy` method by using a symbol for the property name that only the NetworkController has access to? Or maybe we can somehow work out how _not_ to add a `setProxy` method at all. That would be nice.